### DISCUSSION TITLE
Spotify audio quality difference: cspot vs spotty

### DISCUSSION THREAD

**Discussion by nagyrobi** [sle118/squeezelite-esp32#238] [Q&A]
I played the same song from Spotify through both paths:
- LMS with spotty addon -> squeezelite
- cspot directly

They really don't sound the same. cspot sounds quieter than spotty (both at max volume of course). My AT BPHS1 headphones also reproduce less bass when through cspot.
Is maybe some kind of processing involved LMS / squeezelite I may not be aware off? I doubt cspot is doing any processing...

Looking at [cspot doc](https://github.com/feelfreelinux/cspot#architecture):

> An audio sink can optionally implement the void volumeChanged(uint16_t volume) method which is called everytime the user changes the volume (for example via Spotify Connect). If an audio sink implements it it should set softwareVolumeControl to false in its consructor to let cspot know to disable the software volume adjustment. **Properly implementing external volume control (for example via dedicated hardware) will result in a better playback quality since all the dynamic range is used to encode the samples.**

Is this the case in cspot implementation here?

**Comment by philippe44** [comment]
It's probably just volume differences. There is no other processing that I'm aware of in cspot or librespot
